src/flags.h

```diff
@@ -0,0 +1,22 @@
+#ifndef FLAGS_H_
+#define FLAGS_H_
+
+#include "../include/raft.h"
+
+/* This node has the 'Flags' feature. */
+#define RAFT_FLAGS_FLAGS ((raft_flags)1 << 0)
```

Comment:
Is it necessary to know if the other node has the flags feature?

As far as I understand, if a node running this new code receives an AppendEntries result from a legacy node which is not running this new code, then the `raft_append_entries_result->flags` field will set be `0`, and the node running this new code will avoid sending to the legacy node anything that the legacy node will not understand (e.g. InstallSnapshot chunks). On the other hand, if the nodes that sends the AppendEntries result is also running this newer code it will set `raft_append_entries_result->flags` either to `0` or to some value different than `0`. If it sets it to `0` it means that it doesn't support any new feature and it can be treated exactly like the legacy node that didn't send the flags field at all. If it did set some flags, then the receiving node will take advantage of the flags that it knows how to deal with (e.g. InstallSnapshot chunks).

If for some reason we need also to differentiate between the case in which `raft_append_entries_result->flags` is `0` because the node didn't send it at all, and the case where the node set it to `0`, then `raft_append_entries_result->version` can be used (it will be `0` in the former case and `1` in the latter).

Assuming what I'm saying matches the intended mechanics, I believe that `RAFT_FLAGS_FLAGS` could be dropped, and `RAFT_DEFAULT_FEATURE_FLAGS` could be set to 0 (and changed later on when InstallSnapshot chunks are introduced).

Reply:
Yeah, that's true, it was more of a convenience to be able to test it easier in the integration test, otherwise I had to get into more hoops to know if the other node had sent some flags because a 0 value could mean no flags or flags not supported.

Reply:
> Yeah, that's true, it was more of a convenience to be able to test it easier in the integration test, otherwise I had to get into more hoops to know if the other node had sent some flags because a 0 value could mean no flags or flags not supported.

Ah I see.

If the non-test code itself wants to distinguish between no flags and no flags supported it can use `raft_append_entries_result->version` (although I suspect it won't be needed in practice).

For the test code in this specific PR, what do you think to just keep the:

```C
munit_assert_ullong(raft->leader_state.progress[1].feature_flags, ==,
                        RAFT_DEFAULT_FEATURE_FLAGS);
```

assertion below in the `replication/receiveFlags`? It's true that for now it won't distinguish the two cases, but it is still exercising the intended code paths and it distinguish the two cases once we add the first feature flag, so it's a short-lived glitch. On the other had it seems a bit odd to have to support the `RAFT_FLAGS_FLAGS` flag indefinitely for this one test that is only temporarily sub-optimal.

Reply:
Anyway, this is minor, and I can see the rational of both approaches. I'm fine both ways.

Reply:
I'll dump the `RAFT_FLAGS_FLAGS` it's just weird, not really useful and transient anyway.

Reply:
One middle-ground approach could be to leave `RAFT_FLAGS_FLAGS` as it is just for this PR, but override/repurpose its bit by renaming `RAFT_FLAGS_FLAGS` to `RAFT_FLAGS_CHUNKED_INSTALL_SNAPSHOT` (or whatever name) in the follow-up PRs. That way we would use `RAFT_FLAGS_FLAGS` temporarily just for these preparation branches, and replace it with an actual feature once ready.

Unless I'm missing something I don't see us taking advantage of `RAFT_FLAGS_FLAGS` in any way in the future, so we might well get rid of it.